RatTracking/TarkovTools/ITarkovToolsRemoteController.cs

```diff
@@ -41,31 +44,59 @@ public TarkovToolsRemoteController()
 				.Subscribe(_ => SendPong());
 		}
 
-		public async Task ConnectAsync(string sessionId)
+		public Task ConnectAsync(string sessionId)
 		{
 			if (string.IsNullOrEmpty(sessionId))
-				throw new TarkovToolsRemoteControllerException("Invalid ID for remote control");
+				throw new ArgumentException("Invalid session ID", nameof(sessionId));
 
 			_sessionId = sessionId;
 
+			return ConnectInternalAsync();
+		}
+
+		private async Task ConnectInternalAsync()
+		{
 			try
 			{
 				await _client.StartOrFail().ConfigureAwait(false);
 				Send(PayloadType.Connect);
 			}
 			catch (WebsocketException e)
 			{
+				await ForceDisconnectAsync().ConfigureAwait(false);
 				throw new TarkovToolsRemoteControllerException("Unable to connect to Tarkov Tools", e);
 			}
 		}
 
-		public Task DisconnectAsync() => _client.Stop(WebSocketCloseStatus.Empty, string.Empty);
+		public Task DisconnectAsync()
+		{
+			_sessionId = string.Empty;
+			return DisconnectInternalAsync();
+		}
+
+		private Task DisconnectInternalAsync() => _client.Stop(WebSocketCloseStatus.Empty, string.Empty);
 
-		public void OpenItem(string itemId)
+		private Task ForceDisconnectAsync()
 		{
-			if (!_client.IsStarted)
+			// If the WebsocketClient.Start method fails to open web socket then:
+			// - any subsequent WebsocketClient.Start calls do nothing, because IsStarted is true
+			// - any subsequent WebsocketClient.Stop calls do nothing, because IsRunning is false
+			// Workaround: update the IsRunning property via reflection.
+			WebsocketClientIsRunningProperty.Value.SetValue(_client, true);
+			return DisconnectInternalAsync();
+		}
```

Comment:
Was this actually ever the case while you tested? It would be good to solve this without reflection if possible but im not sure how to cause the described state so cant really test alternatives.

Reply:
During a couple of week I faced WS connection issue one or two times. I was able to reproduce it by terminating my internet connection (`ipconfig /release`) right before `TarkovToolsRemoteController.ConnectAsync` call and restoring it back (`ipconfig /renew`) after.

I dont like reflection solution either, but as the comment states looks like it's only the option.